pinot-core/src/main/java/com/linkedin/pinot/core/startree/StarTreeIndexNodeV2.java

```diff
@@ -0,0 +1,326 @@
+  private int childrenEndIndex;
+
+  /**
+   * Constructor for the class.
+   * - Reads all fields from the data buffer.
+   *
+   * @param dataBuffer
+   * @param nodeId
+   */
+  public StarTreeIndexNodeV2(LBufferAPI dataBuffer, int nodeId) {
+    this.dataBuffer = dataBuffer;
+    long offset = nodeId * size;
+
+    dimensionName = dataBuffer.getInt(offset);
+    offset += Integer.BYTES;
+
+    dimensionValue = dataBuffer.getInt(offset);
+    offset += Integer.BYTES;
+
+    startDocumentId = dataBuffer.getInt(offset);
+    offset += Integer.BYTES;
+
+    endDocumentId = dataBuffer.getInt(offset);
+    offset += Integer.BYTES;
+
+    aggregatedDocumentId = dataBuffer.getInt(offset);
+    offset += Integer.BYTES;
+
+    childrenStartIndex = dataBuffer.getInt(offset);
+    offset += Integer.BYTES;
+
+    childrenEndIndex = dataBuffer.getInt(offset);
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public int getDimensionName() {
+    return dimensionName;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @param dimensionName
+   */
+  @Override
+  public void setDimensionName(int dimensionName) {
+    this.dimensionName = dimensionName;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public int getDimensionValue() {
+    return dimensionValue;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @param dimensionValue
+   */
+  @Override
+  public void setDimensionValue(int dimensionValue) {
+    this.dimensionValue = dimensionValue;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public int getStartDocumentId() {
+    return startDocumentId;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @param startDocumentId
+   */
+  @Override
+  public void setStartDocumentId(int startDocumentId) {
+    this.startDocumentId = startDocumentId;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public int getEndDocumentId() {
+    return endDocumentId;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @param endDocumentId
+   */
+  @Override
+  public void setEndDocumentId(int endDocumentId) {
+    this.endDocumentId = endDocumentId;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public int getAggregatedDocumentId() {
+    return aggregatedDocumentId;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @param aggregatedDocumentId
+   */
+  @Override
+  public void setAggregatedDocumentId(int aggregatedDocumentId) {
+    this.aggregatedDocumentId = aggregatedDocumentId;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public int getNumChildren() {
+    return (childrenEndIndex != INVALID_INDEX) ? (childrenEndIndex - childrenStartIndex + 1) : 0;
+  }
+
+  /**
+   * {@inheritDoc}
+   * @return
+   */
+  @Override
+  public boolean isLeaf() {
+    return (getNumChildren() == 0);
+  }
+
+  /**
+   * {@inheritDoc}
+   * @param child
+   * @param dimensionValue
+   */
+  @Override
+  public void addChild(StarTreeIndexNodeInterf child, int dimensionValue) {
```

Comment:
Remove this method from this class and the interface since it's version dependent.